### PR TITLE
Alow to select a system to go directly to the game list (Fixes #253)

### LINKED ISSUE #253: Alow to select a system to go directly to the game list
When we have just one platform with ROMs in the system, we are still presented to the platform selection screen.

The improvement is to check when this happens and skips the platform selection screen, going directly to that platform ROMs list.

### PR DESCRIPTION
**Pull request by raelgc**
Fixes #253

### CODE REVIEW

**Comment by raelgc** [comment]
Thanks for the comments, @joolswills. 

I've created #253 for this purpose (discuss the idea) :)

If you think this is not acceptable before further discussion, you can close this PR and we can discuss in the ticket.

**Comment by joolswills** [comment]
Leave this for now - but it will need at least the changes I mentioned, and I would like to hear the views of others though on the ticket. If agreed the PR can be updated etc.

**Comment by raelgc** [comment]
> it needs to be at least configurable and it should exclude the RetroPie system.

A command line config param for the `Settings` class will be enough?

> it should exclude the RetroPie system.

Sorry, I didn't understand this one.

**Comment by pjft** [comment]
@raelgc There's a "RetroPie" system in the carousel. I think that's what @joolswills is referring to?

In terms of a command line config param, could you elaborate? On the Pi, users don't really use command-line parameters often. I imagine @joolswills was referring to an option in the menu, for users to be able to select that - similar to the "quick navigation" option.

**Comment by joolswills** [comment]
the option needs to be in the GUI and default to current behaviour.

We have a RetroPie system which we wouldn't want this feature to work on else on first boot it would go right to the configuration options.

**Comment by raelgc** [comment]
@joolswills Thanks for the answers! Indeed, all makes sense.

Just to answer @pjft about the command line args: my initial thought was to use command line args similar to `--no-splash` present in [`parseArgs`](https://github.com/RetroPie/EmulationStation/blob/master/es-app/src/main.cpp#L33).

**Comment by raelgc** [comment]
@joolswills I just changed to check if the only present system isGameSystem or not. I tested with only one game system and it gone directly to the game system. Then tested with only Retropie as the only system, then the initial screen was not scaped.

**Comment by raelgc** [comment]
Or should I check if `"retropie" == system->getName()` to allow other systems, like Kodi, to load directly if it's the only one present?

**Comment by raelgc** [comment]
Ok guys, I changed the way it works. It'll skip the system selector screen only if an optional command line argument `system` is passed. 

Of course, it excludes `retropie`.

Examples: 

1. `emulationstation --system mastersystem` will skip system selector after load and go directly to the `mastersystem` gamelist (if exists, of course).

2. `emulationstation --system retropie` will not work and system selector screen is loaded.

3. `emulationstation --system foobarxyz` will not work and system selector screen is loaded.

**Comment by joolswills** [comment]
It needs to be in the GUI as I mentioned before. Thanks.

**Comment by raelgc** [comment]
@joolswills, now as GUI option under UI Settings. `None` will be current behavior. `retropie` is not in the list.

File: es-app/src/views/ViewController.cpp [hunk 1]
```diff
@@ -503,7 +511,7 @@ void ViewController::monitorUIMode()
 {
 	std::string uimode = Settings::getInstance()->getString("UIMode");
 	if (uimode != mCurUIMode) // UIMODE HAS CHANGED
-	{	
+	{
```
Comment (zigurana) [inline es-app/src/views/ViewController.cpp]: thanks for the cleanup

File: es-app/src/views/ViewController.cpp [hunk 2]
```diff
@@ -44,6 +44,18 @@ ViewController::~ViewController()
 
 void ViewController::goToStart()
 {
+	// If specific system is requested, go directly to the game list (Fixes #253)
```
Comment (zigurana) [inline es-app/src/views/ViewController.cpp]: No need to reference the issue #253 in a comment here.

File: es-core/src/Settings.cpp [hunk 3]
```diff
@@ -16,6 +16,7 @@ std::vector<const char*> settings_dont_save = boost::assign::list_of
 	("ForceKiosk")
 	("IgnoreGamelist")
 	("HideConsole")
+	("System")
```
Comment (zigurana) [inline es-core/src/Settings.cpp]: If we want this setting to persist across sessions (power-down/ups), this needs to be removed.

File: es-core/src/Settings.cpp [hunk 4]
```diff
@@ -47,6 +48,7 @@ void Settings::setDefaults()
 	mBoolMap["ShowExit"] = true;
 	mBoolMap["Windowed"] = false;
 	mBoolMap["SplashScreen"] = true;
+	mStringMap["System"] = "";
```
Comment (zigurana) [inline es-core/src/Settings.cpp]: I think the name ` System` should be more descriptive. Maybe `StartupSystem`, or even `SystemToShowAtStartup` ?

File: es-app/src/main.cpp [hunk 5]
```diff
@@ -65,6 +65,9 @@ bool parseArgs(int argc, char* argv[], unsigned int* width, unsigned int* height
 		}else if(strcmp(argv[i], "--no-splash") == 0)
 		{
 			Settings::getInstance()->setBool("SplashScreen", false);
+		}else if(strcmp(argv[i], "--system") == 0)
```
Comment (zigurana) [inline es-app/src/main.cpp]: Do we want to retain the command line option on top of the UI-based setting?
It does not hurt of course...

**Comment by raelgc** [comment]
@zigurana Thanks for the review. All changes done!

**Comment by raelgc** [comment]
@joolswills Now as UI entry only.

**Comment by joolswills** [comment]
Please can you squash the commits and I will merge thanks. Note - whitespace fixes are generally better in a separate commit. thanks.

**Comment by raelgc** [comment]
@joolswills I squashed all changes in commit https://github.com/RetroPie/EmulationStation/pull/257/commits/b1f892244e7362325580066b237c28b738d106a1

Thank you for the whitespace fix tip.

**Comment by raelgc** [comment]
@joolswills Hope the above squashed commit is enough. If not, should I close this PR and create a new one with only the squashed commit?

**Comment by zigurana** [comment]
Hey @raelgc, let me try to explain what needs to happen.
In order for the commit history of the master branch to be somewhat readable over time, it's preferred to squash all commits into a single one. 
That can be done in multiple ways:
`git rebase --interactive [retropie/master] raelgc:master` where  [retropie/master] is your local reference to that stream (depends on how you configured your remotes). The keyword `interactive` will open up a list of all commits, which you can then be combined into a single commit. In your current case you could leave out the commits that do not add anything (the last two). 
After completing this procedure, you will probably be unable to push to raelgc:master, because git might think you have later commits on the server (it is unaware of your rebase activity, which removed those later ones). If so, just do:
`git push --force master [remoteBranch]:[localBranch]`

See: https://git-scm.com/docs/git-rebase
See also this reference: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

Another way to achieve this is by stashing your current state, rewind the branch 10 commits, apply the stash and commit. that should do the same thing (I think).

The thing you do not need to do, is to close this PR, and open a new one.

**Comment by raelgc** [comment]
@zigurana Thanks for the instructions.

Indeed, I've made the second way, but it just produced that single commit (and kept others).

I'll try with `rebase`.

**Comment by raelgc** [comment]
@zigurana and @joolswills now properly squashed in a single commit. Thank you for all explanations and your patience.

**Pull request merged**